### PR TITLE
Allow to add and set HTTP headers using `java.util.Map`

### LINKED ISSUE #3499: Scala extension method that converts `Map[CharSequence, String]` to `Iterator<Entry<>>`
It is currently quite inconvenient when a Scala user tries to put multiple headers into `HttpHeadersBuilder` from a `Map`:

```scala
val headers: Map[String, String] = ...
val builder = HttpHeaders.builder()
headers.foreach { tuple => builder.add(tuple._1, tuple._2) }
```

It'd be nice if we add an extension method to `Iterable[(CharSequence, String)]` (and `Iterable[(String, String)]`?) so it's easier to convert it into an `Iterator[Map.Entry[CharSequence, String]]`:

```
builder.add(headers.mapEntryIterator)
```

### PR DESCRIPTION
**Pull request by ikhoon**
Motivation:

`HttpHeadersBuilder` only take `Iterable<? extends Entry<? extends CharSequence, String>>` type,
so users have to convert `Map<String, String>` to Iterable<Entry<String, String>>` manually using
`map.entrySet()`.
It'd be convenient to add or set directly from `Map`

Modifications:

- Add shortcut methods for add Http headers using Map type
 - `set(Map<? extends CharSequence, String>)`
 - `add(Map<? extends CharSequence, String>)`

Result:

- You can now add or set HTTP headers using `java.util.Map` type.
- Fixes #3499

### CODE REVIEW

File: core/src/main/java/com/linecorp/armeria/common/HttpHeadersBuilder.java [hunk 1]
```diff
@@ -230,6 +231,15 @@
      */
     HttpHeadersBuilder add(Iterable<? extends Entry<? extends CharSequence, String>> entries);
 
+    /**
+     * Adds all header names and values of the specified {@code entries}.
+     *
+     * @return {@code this}
+     */
+    default HttpHeadersBuilder add(Map<? extends CharSequence, String> entries) {
+        return add(entries.entrySet());
```
Comment (minwoox) [inline core/src/main/java/com/linecorp/armeria/common/HttpHeadersBuilder.java:240]: nit: rnn?

File: core/src/main/java/com/linecorp/armeria/common/QueryParamsBuilder.java [hunk 2]
```diff
@@ -228,6 +229,15 @@
      */
     QueryParamsBuilder add(Iterable<? extends Entry<? extends String, String>> entries);
 
+    /**
+     * Adds all parameter names and values of the specified {@code entries}.
+     *
+     * @return {@code this}
+     */
+    default QueryParamsBuilder add(Map<String, String> entries) {
+        return add(entries.entrySet());
```
Comment (minwoox) [inline core/src/main/java/com/linecorp/armeria/common/QueryParamsBuilder.java:238]: ditto

**Review by minwoox — APPROVED** [review]
Thanks! 😄

File: core/src/main/java/com/linecorp/armeria/common/HttpHeadersBuilder.java [hunk 3]
```diff
@@ -377,6 +387,17 @@
...
 
+    /**
+     * Retains all current headers but calls {@link #set(CharSequence, String)} for each header in
+     * the specified {@code entries}.
+     *
+     * @param entries the headers used to set the header values
+     * @return {@code this}
+     */
+    default HttpHeadersBuilder set(Map<? extends CharSequence, String> entries) {
+        return set(entries.entrySet());
```
Comment (trustin) [inline core/src/main/java/com/linecorp/armeria/common/HttpHeadersBuilder.java:398]:
```suggestion
        return set(requireNonNull(entries, "entries").entrySet());
```

**Review by trustin — APPROVED** [review]
Thanks!

**Pull request merged**